app/decorators/check-auth.ts

```diff
@@ -48,7 +48,7 @@ export default function checkAuth<T extends ConcreteSubclass<Route>>(
                         'error-no-api',
                         transitionTargetURL(transition).slice(1),
                     );
-                    return;
+                    return undefined;
```

Comment:
What's the reason for returning undefined everywhere? We should try to leave this as implicit as possible.

Reply:
[consistent-return](https://eslint.org/docs/rules/consistent-return)

This was already enabled in [eslint-config-airbnb-base@12.1.0](https://github.com/airbnb/javascript/blob/eslint-config-airbnb-base-v12.1.0/packages/eslint-config-airbnb-base/rules/best-practices.js#L23)

It's not clear to me why this was not being enforced previously.

FWIW, I tried setting `treatUndefinedAsUnspecified: true` (because the description of it is a little confusing), but this means something else (exactly what is says, actually).

Reply:
We should refactor these functions as to where `return undefined` is unnecessary. Also, how is this not failing on the first if() statement?

Reply:
Not sure what you mean. How would you expect it to fail?

Reply:
I'll provide an example.

Reply:
```suggestion
                } else if (transition.targetName !== 'error-no-api') {
                    // Must have failed to make the request at all.
                    this.transitionTo(
                        'error-no-api',
                        transitionTargetURL(transition).slice(1),
                    );
                    return;
                }
            }
            // In any case, let the route load normally.
            super.beforeModel(transition);
```

Simple version.

Reply:
```javascript
export default function checkAuth<T extends ConcreteSubclass<Route>>(
    RouteSubclass: T,
) {
    class AuthenticatedRoute extends RouteSubclass {
        @service router!: ServiceRegistry['router'];
        @service session!: SessionService;

        checkUserAuth = async (transition: any) => {
            try {
                if (!this.session.isAuthenticated) {
                    await this.session.authenticate(authenticator);
                }
            } catch (e) {
                if (e instanceof NotLoggedIn) {
                    return 'NOT_LOGGED_IN';
                }
                if (transition.targetName !== 'error-no-api') {
                    return 'LOGIN_REQUEST_FAILED';
                }
            }
            return 'LOGIN_COMPLETE';
        }

        async beforeModel(transition: any) {
            const auth = await this.checkUserAuth(transition);

            switch (auth) {
            case 'NOT_LOGGED_IN':
                this.session.on('authenticationSucceeded', this, this.refresh);
                break;
            case 'LOGIN_REQUEST_FAILED':
                this.transitionTo(
                    'error-no-api',
                    transitionTargetURL(transition).slice(1),
                );
                break;
            default:
                super.beforeModel(transition);
            }
        }
    }

    return AuthenticatedRoute;
}
```
Better version w/overall refactor